idunn/api/instant_answer.py

```diff
@@ -127,16 +127,30 @@ async def get_instant_answer(
 
     if not intentions:
         bragi_response = await bragi_client.raw_autocomplete(
-            {"q": normalized_query, "lang": lang, "limit": 1}
+            {"q": normalized_query, "lang": lang, "limit": 5}
         )
-        if len(bragi_response["features"]) > 0:
-            place_geocoding = bragi_response["features"][0]["properties"]["geocoding"]
-            place_id = place_geocoding["id"]
-            if result_filter.check_bragi_response(normalized_query, place_geocoding):
-                result = await run_in_threadpool(
-                    get_instant_answer_single_place, place_id=place_id, lang=lang
-                )
-                return build_response(result, query=q, lang=lang)
+
+        geocodings = sorted(
+            (
+                (rank, feature["properties"]["geocoding"])
+                for feature in bragi_response["features"]
+                for rank in [
+                    result_filter.rank_bragi_response(
+                        normalized_query, feature["properties"]["geocoding"]
+                    )
+                ]
+                if rank is not None
+            ),
+            key=lambda item: -item[0],  # sort by descending rank
+        )
```

Comment:
As discussed, this `rank_bragi_response` may take several milliseconds in some cases.  
I would suggest to ignore this reranking step when the first result is neither a street or an address, as the first result will not change anyway.